backend/ecs_tasks/delete_files/json_handler.py

```diff
@@ -0,0 +1,66 @@
+import gzip
+import io
+import json
+import logging
+from collections import Counter
+
+import pyarrow as pa
+
+logger = logging.getLogger(__name__)
+
+
+def initialize(input_file, out_stream, compressed):
+    if compressed:
+        bytestream = io.BytesIO(input_file.read())
+        input_file = gzip.GzipFile(None, "rb", fileobj=bytestream)
+    gzip_stream = pa.CompressedOutputStream(out_stream, "gzip") if compressed else None
+    writer = gzip_stream if compressed else out_stream
+    return input_file, writer
+
+
+def find_key(key, obj):
+    """
+    Athena openx SerDe is case insensitive, and converts by default each object's key
+    to a lowercase value: https://docs.aws.amazon.com/athena/latest/ug/json-serde.html
+
+    Here we convert the DataMapper value for the column identifier
+    (for instance, customerid) to the JSON's object key (for instance, customerId).
+    """
+    if not obj:
+        return None
+    for found_key in obj.keys():
+        if key.lower() == found_key.lower():
+            return found_key
+
+
+def delete_matches_from_json_file(input_file, to_delete, compressed=False):
+    deleted_rows = 0
+    with pa.BufferOutputStream() as out_stream:
+        input_file, writer = initialize(input_file, out_stream, compressed)
+        content = input_file.read().decode("utf-8")
+        lines = content.split("\n")
+        if lines[-1] == "":
+            lines.pop()
+        total_rows = len(lines)
+        for line in lines:
+            parsed = json.loads(line)
```

Comment:
one small issue here is that if a row in the JSON file contains invalid JSON, a JSONDecodeError will be thrown (subclass of ValueError) which will end up being caught in the ValueError block in `main.py` and the message might be confusing because it'll say something like "Unprocessable message: some error about JSON structure" however you can't tell from that error whether it's a problem with the message which was placed on the queue, or the file itself. You could tellby looking at the container logs since you could see from other log statements that the queue message was read OK, but on the error event it wouldn't be clear that the data was messed up. Perhaps we should handle decode errors here an rethrow a value error but with a more helpful error message

Reply:
Done - I tried to construct a meaningful sentence out of the error and added a test. 
In general, we have to consider that Athena should possibly blow if there's any malformed JSON as we don't allow the ignore option, so this is possibly a edge case that would be in need of being dealt with only if you put a new object between Find and Forget, but I think it's a good call, allows better troubleshooting!